### PR TITLE
Red Chapter 17 Answers

### PR DESCRIPTION
**Pull request by utopia-2**

### CODE REVIEW

File: answers.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      and the `Translate` transform._
+
+See [EffectsDemo.java](src/org/java/training/EffectsDemo.java).
+
+ 16) _Continue to advance in your knowledge of Java. A good way to start is by examining Java's core
+      packages, such as `java.lang`, `java.util`, and `java.net`. Write sample programs
+      that demonstrate their various classes and interfaces. In general, the best way to become a
+      great Java programmer is to write lots of code._
+
+It is done (/I'll just move onto some Rust instead).
```
Comment (alexdeas) [inline answers.md:85]: Congratulations! If you have read and worked through the preceding 17 chapters, then you can call yourself a Java programmer.

File: src/org/java/training/BlurryRotator.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+public class BlurryRotator extends Application {
+
+	double angle = 0.0, scaleFactor = 0.4, blurAmount = 1.0;
+	Reflection reflect = new Reflection();
+	Rotate rotate = new Rotate();
+	Scale scale = new Scale(scaleFactor, scaleFactor);
+	BoxBlur
+		blur = new BoxBlur(1.0, 1.0, 1),
+		rotateBlur = new BoxBlur(5.d, 5.d, 2);
+
```
Comment (alexdeas) [inline src/org/java/training/BlurryRotator.java:24]: This looked weird until I realised you were instantiating objects on an extended line.

File: src/org/java/training/FileComparison.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		Button compare = new Button("Compare");
+		compare.setOnAction(e -> compare(a.getText(), b.getText(), result::setText));
+
+		Arrays.asList(a, b).forEach(tf -> tf.setOnAction(e -> compare.fire()));
+
+		root.getChildren().addAll(a, b, compare, result);
+		stage.show();
+	}
+
+	public void compare(String first, String second, Consumer<String> out) {
```
Comment (alexdeas) [inline src/org/java/training/FileComparison.java:41]: `result::setText` is also a `Consumer<String>`?

Comment (utopia-2) [inline src/org/java/training/FileComparison.java:41]: `result::setText` is whatever you want it to be if it conforms to the signature of the target functional interface. This is what makes method references awesome.

File: src/org/java/training/BlurryRotator.java (hunk 2 above)
Comment (utopia-2) [inline src/org/java/training/BlurryRotator.java:24]: Yeah, I like to group instantiation like this. It visually separates groups of related instances quite nicely.

File: answers.md (hunk 1 above)
Comment (utopia-2) [inline answers.md:85]: I didn't do chapter 15, so I guess I can't call myself a Java programmer just yet.

**Pull request closed without merging**